cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/shared_links.spec.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+import {
+  DX_DEFAULT_END_TIME,
+  DX_DEFAULT_START_TIME,
+} from '../../../../../utils/constants';
+
+const miscUtils = new MiscUtils(cy);
+
+describe('test large strings', () => {
+  before(() => {
+    cy.setAdvancedSetting({
+      defaultIndex: 'logstash-*',
+    });
+
+    miscUtils.visitPage('app/data-explorer/discover#/');
+    cy.waitForLoader();
+    cy.setTopNavDate(DX_DEFAULT_START_TIME, DX_DEFAULT_END_TIME);
+    cy.waitForSearch();
+  });
+
+  describe('shared links with state in query', () => {
+    it('should allow for copying the snapshot URL', function () {
+      const url = `http://localhost:5601/app/data-explorer/discover#/?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(indexPattern:'logstash-*',view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2015-09-19T13:31:44.000Z',to:'2015-09-24T01:31:44.000Z'))&_q=(filters:!(),query:(language:kuery,query:''))`;
```

Comment:
haven't ran this one yet. this doesn't cause a failure when it includes an extra base path right?

Reply:
Ive fixed that issue in 2.10, so no :)